js/base/Exchange.js

```diff
@@ -2118,7 +2125,7 @@ module.exports = class Exchange {
         };
     }
 
-    safeMarket (marketId = undefined, market = undefined, delimiter = undefined) {
+    safeMarket (marketId = undefined, market = undefined, delimiter = undefined, marketType = 'spot') {
```

Comment:
I like a lot the approach.
Would it make sense to have `marketType = undefined` and then check inside if `marketType === undefined` use defaultType ?

Reply:
@frosty00 I agree with Pablo, I think we should fallback to the `defaultType`

Reply:
also `contract` (used below) is not a unified market type, if we only care about the spot vs derivatives conflicts, why not make this a bool (`isSpot`) ? seems less error-prone to me